chain/state/statetree.go

```diff
@@ -115,17 +117,18 @@ func (ss *stateSnaps) deleteActor(addr address.Address) {
 	ss.layers[len(ss.layers)-1].actors[addr] = streeOp{Delete: true}
 }
 
-func NewStateTree(cst cbor.IpldStore) (*StateTree, error) {
+func NewStateTree(cst cbor.IpldStore, version network.Version) (*StateTree, error) {
 
 	return &StateTree{
-		root:  adt.MakeEmptyMap(adt.WrapStore(context.TODO(), cst)),
+		root:  adt.NewMap(adt.WrapStore(context.TODO(), cst), version),
 		Store: cst,
 		snaps: newStateSnaps(),
 	}, nil
 }
 
-func LoadStateTree(cst cbor.IpldStore, c cid.Cid) (*StateTree, error) {
-	nd, err := adt.AsMap(adt.WrapStore(context.TODO(), cst), c)
+func LoadStateTree(cst cbor.IpldStore, c cid.Cid, version network.Version) (*StateTree, error) {
+	// NETUPGRADE: switch map adt type on version upgrade.
+	nd, err := adt.AsMap(adt.WrapStore(context.TODO(), cst), c, version)
```

Comment:
Using the adt wrappers here is not strictly necessary - the underlying hamt could be accessed instead. That's still versioned, of course, but probably changing less frequently.

Reply:
It's actually easier to just do it this way as we need to support multiple HAMT versions anyways.